ndcube/asdf/converters/tests/test_ndcube_converter.py

```diff
@@ -0,0 +1,43 @@
+import numpy as np
+import pytest
+
+import asdf
+
+from ndcube.tests.helpers import assert_cubes_equal
+
+
+@pytest.mark.parametrize("ndc",[("ndcube_gwcs_2d_ln_lt"),
+                                ("ndcube_gwcs_3d_ln_lt_l"),
+                                ("ndcube_gwcs_3d_ln_lt_l_ec_gc"),
+                                ("ndcube_gwcs_3d_rotated"),
+                                ("ndcube_gwcs_4d_ln_lt_l_t"),
+                                ], indirect=("ndc",))
+def test_serialization(ndc, tmp_path):
+    file_path = tmp_path / "test.asdf"
+    with asdf.AsdfFile() as af:
+        af["ndcube_gwcs"] = ndc
+        af.write_to(file_path)
+
+    with asdf.open(file_path) as af:
+        assert_cubes_equal(af["ndcube_gwcs"], ndc)
+
+@pytest.mark.xfail(reason="Serialization of sliced ndcube not supported")
```

Comment:
@braingram I have used `@pytest.mark.xfail` to indicate that other `NDCube` objects are not supported for reading and writing to an ASDF file. If you have any suggestions on how to better inform about the unsupported objects, please let me know.

Reply:
I think this is a great way to handle it.